plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/jdbc/TrinoJdbcCatalog.java

```diff
@@ -186,13 +190,15 @@ public Transaction newCreateTableTransaction(ConnectorSession session, SchemaTab
     @Override
     public void registerTable(ConnectorSession session, SchemaTableName tableName, String tableLocation, String metadataLocation)
     {
-        throw new TrinoException(NOT_SUPPORTED, "registerTable is not supported for Iceberg JDBC catalogs");
+        jdbcClient.createTable(tableName.getSchemaName(), tableName.getTableName(), metadataLocation);
```

Comment:
should we add the reason behind not using the jdbc catalogs' register table (seems like `Cannot invoke "org.apache.iceberg.util.SerializableSupplier.get()" because "this.hadoopConf" is null`) so that someone doesn't try to "optimize" this by removing jdbcClient object that's being used for this specific purpose? Or when the underlying code gets fixed - we can update this.